libsigner/src/events.rs

```diff
@@ -345,8 +346,10 @@ fn process_stackerdb_event(
             .filter_map(|chunk| read_next::<NakamotoBlock, _>(&mut &chunk.data[..]).ok())
             .collect();
         SignerEvent::ProposedBlocks(blocks)
-    } else if event.contract_id.name.to_string() == SIGNERS_NAME {
-        // TODO: fix this to be against boot_code_id(SIGNERS_NAME, is_mainnet) when .signers is deployed
+    } else if event.contract_id.name.to_string().starts_with(SIGNERS_NAME)
+        && event.contract_id.issuer.1 == [0u8; 20]
+    {
```

Comment:
I think we might need some sort of additional item passed through with the SignerMessage to indicate whether this came from an even or odd rewrd cycle number so the signer knows which state machine to feed it to. I would assume the state machine would just ignore it as it isn't signed correctly, but we could have the case where a signer is signing for both reward cycle N and N+1 and therefore produces a valid signature and the receiving signer feed them into both state machines will bork up one of them.

Reply:
Yeah, I was thinking about that when I wrote it, but I didn't know what the best way to proceed was

Reply:
I think I have a fix in my PR for this so no need to add :D